optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc

```diff
@@ -57,3 +57,86 @@ and supports <<explainingTheScore,justifications>>:
                         Talk::getDurationInMinutes);
     }
 ----
+
+[[constraintStreamsGroupingAndCollectors]]
+== Grouping and Collectors
```

Comment:
in this section, I'd make 2 subsections, besides the introduction with a first example:
- group key's
- collectors

That way, all the build-in collectors, are grouping by a single section, separate from the group key one